proto/workspace.proto

```diff
@@ -0,0 +1,140 @@
+syntax = "proto3";
+
+import "proto/context.proto";
+
+package workspace;
+
+message GetWorkspaceRequest {
+  // The request context.
+  context.RequestContext request_context = 1;
+
+  // The name of the workspace to fetch.
+  string name = 2;
+}
+
+message GetWorkspaceResponse {
+  // The response context.
+  context.ResponseContext response_context = 1;
+
+  // The requested workspace.
+  // To read workspace file contents, use the `GetWorkspaceFile` endpoint.
+  // To read workspace directory contents, use the
+  // `GetWorkspaceDirectoryRequest` endpoint.
+  Workspace workspace = 2;
+}
+
+message SaveWorkspaceRequest {
+  // The request context.
+  context.RequestContext request_context = 1;
+
+  // The workspace to save.
+  Workspace workspace = 2;
+}
+
+message SaveWorkspaceResponse {
```

Comment:
100% not necessary, but you could return name here if you wanted to more easily match req/rsp

Reply:
Good idea, didn't think about matching these up. Done.